src/contents/shared/ProfileEdit.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import { Component } from 'react';
+import fire from '../../config/Fire';
+import { ProfileEditView } from './ProfileEditView';
+import firebase from 'firebase';
+import { storage } from '../../config/Fire';
+
+export class ProfileEdit extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+		  image: "https://example.com/jane-q-user/profile.jpg",
+		  url: this.props.user.photoURL,
+		  progress: 0,
+		  name: this.props.user.displayName,
+		  email: this.props.user.email,
+		  type: this.props.user.type,
+		  location: this.props.user.location,
+		  photoURL: this.props.user.url,
+		}
+		this.handleChangeUpload = this
+      .handleChangeUpload
+      .bind(this);
+       
+	  }
+
+
+	database = fire.database().ref();
+
+	/**
+		* Updates state to the current value of a certain target.
+		* @param e
+		*/
+		handleChange = e => {
+			this.setState({ [e.target.name]: e.target.value });
+		};
+
+	/**
+		* Registers the user as photographer or company.
+		*
+		* @param e
+		*/
+
+	updateUser = (e) => {
+		e.preventDefault();
+		let { name, email, type, location, url } = this.state;
+		const { user } = this.props;
+
+		if (name !== "" && email !== "" && location !== "" && url !== "") {
+
+		var newUser = firebase.auth().currentUser;
+
+		newUser.updateProfile({
+
+			displayName: name,
+			photoURL: url
+		})
+			
+			.then(() => {
+
+				if(user.location !== location && location !=="") {
+					this.database.child("locations").child(user.location).child(user.type).child(user.uid).remove();
+					
+					
+					this.database.child("locations").child(location).child(user.type).child(user.uid).update({
+						displayName: name,
+						photoURL: url
+					});
+				}
+				else {
+					console.log("location was same as before so no update:" + user.location);
+
+				}
+
+			})
+
+			.then(() => {
+				this.database.child("users").child(user.uid).update({
+					type: type,
+					email: email,
+					location: location,
+					displayName: name,
+					photoURL: url
+				});
+
+			})
+
+			.then(() => {
+				this.database.child(user.type).child(user.uid).update({
+					email: email,
+					location: location,
+				});
+			})
+
+			.then(() => {
+				alert("User Updated");
+			})	
+	
+		}
+		else {
+			alert("please fill in all the input fields");
+		}
+	};
+
+	handleChangeUpload = e => {
+		this.setState({ [e.target.name]: e.target.value });
```

Comment:
Dragos also implemented the profile picture upload, please compare your version to his.

Reply:
Done! i have used his component